pickui/src/main/kotlin/com/lovoo/android/pickui/adapter/PopUpGalleryAdapter.kt

```diff
@@ -45,32 +44,30 @@ class PopUpGalleryAdapter(
 
     private val inflater = LayoutInflater.from(context)
 
-    override fun getView(position: Int, convertView: View?, container: ViewGroup?): View {
+    override fun getView(
+        position: Int,
+        convertView: View?,
+        container: ViewGroup
+    ): View {
         val item = getItem(position) ?: return View(context)
-        val view = convertView ?: inflater.inflate(R.layout.pickpic_list_item_gallery, container, false)
-
-        view.gallery_cover.viewTreeObserver.addOnPreDrawListener(object : ViewTreeObserver.OnPreDrawListener {
-            override fun onPreDraw(): Boolean {
-                view.gallery_cover.viewTreeObserver.removeOnPreDrawListener(this)
-                val file = File(item.coverPath)
-                val uri = when (file.exists()) {
-                    true -> Uri.fromFile(file)
-                    false -> Uri.parse(item.coverPath)
+        return convertView ?: PickpicListItemGalleryBinding.inflate(inflater, container, false).apply {
```

Comment:
@Nidvex convertView will never be updated

everything empty
-> create item for DCIM folder

scrolling down
-> the view from DCIM is reused for a new entry like Snapshots

but since the view is never updated it will show DCIM instead of Snapshot

please extract binding 
`val binding = convertView?.let { PickpicListItemGalleryBinding.bind(it) } ?: PickpicListItemGalleryBinding.inflate(inflater, container, false)`

`return binding.apply {...}.root`